src/test/kotlin/io/github/mojira/arisa/modules/CrashInfoModuleTest.kt

```diff
@@ -0,0 +1,197 @@
+            resolveAsInvalid = { Unit.right() },
+            resolveAsDuplicate = { Unit.right() },
+            createLink = { _, _, _ -> Unit.right() },
+            addComment = { Unit.right() }
+        )
+
+        val result = module(issue, A_SECOND_AGO)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should return OperationNotNeededModuleResponse when issue does not contain any recent crash as attachment" {
+        val module = CrashInfoModule(
+            listOf("txt"),
+            crashReader
+        )
+
+        val attachment = getAttachment(
+            content = PIXEL_FORMAT_CRASH,
+            created = NOW.minus(42, ChronoUnit.DAYS)
+        )
+
+        val issue = mockIssue(
+            attachments = listOf(attachment),
+            description = "",
+            created = NOW,
+            confirmationStatus = Unconfirmed,
+            resolveAsInvalid = { Unit.right() },
+            resolveAsDuplicate = { Unit.right() },
+            createLink = { _, _, _ -> Unit.right() },
+            addComment = { Unit.right() }
+        )
+
+        val result = module(issue, A_SECOND_AGO)
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should return OperationNotNeededModuleResponse when attached crash does have a wrong mime type" {
+        val module = CrashInfoModule(
+            listOf("txt"),
+            crashReader
+        )
+
+        val attachment = getAttachment(
+            name = "crash.png",
+            content = PIXEL_FORMAT_CRASH
+        )
+        val issue = mockIssue(
+            attachments = listOf(attachment),
+            description = "",
+            created = NOW,
+            confirmationStatus = Unconfirmed,
+            resolveAsInvalid = { Unit.right() },
+            resolveAsDuplicate = { Unit.right() },
+            createLink = { _, _, _ -> Unit.right() },
+            addComment = { Unit.right() }
+        )
+
+        val result = module(issue, A_SECOND_AGO)
+
+        result.shouldBeLeft(OperationNotNeededModuleResponse)
+    }
+
+    "should add attachment when deobfuscated" {
+        var resolvedAsDupe = false
+        var resolvedAsInvalid = false
+        var addedAttachement = false
+
+        val module = CrashInfoModule(
+            listOf("txt"),
+            crashReader
+        )
+        val issue = mockIssue(
+            attachments = listOf(getAttachment(OBFUSCATED_CRASH)),
+            description = OBFUSCATED_CRASH,
+            created = NOW,
+            confirmationStatus = Unconfirmed,
+            resolveAsDuplicate = { resolvedAsDupe = true },
+            resolveAsInvalid = { resolvedAsInvalid = true },
+            addAttachment = { addedAttachement = true }
+        )
+
+        val result = module(issue, A_SECOND_AGO)
+
+        result.shouldBeRight(ModuleResponse)
+        resolvedAsDupe.shouldBeFalse()
+        addedAttachement.shouldBeTrue()
+        resolvedAsInvalid.shouldBeFalse()
+    }
+})
```

Comment:
No tests for the comment that this PR adds?

Reply:
Added tests.